.github/scripts/update_python_dependencies.sh

```diff
@@ -0,0 +1,23 @@
+#! /bin/sh
+set -e
+
+# Check for required arguments
+if [ "$#" -ne 2 ]; then
+    echo "Usage: update_python_dependencies [environment_name] [target]"
+    exit 1
+fi
+ENV_NAME=$1
+TARGET=$2
+
+# Check for pip-compile
+if ! command -v pip-compile > /dev/null; then
+    echo "pip-compile could not be found"
+    exit 1
+fi
+
+# Run pip-compile
+if [ "$ENV_NAME" = "default" ]; then
+    pip-compile -U pyproject.toml -o "$TARGET"
+else
+    hatch env show --json | jq -r ".${ENV_NAME}.dependencies | .[]" | pip-compile -U - -o "$TARGET"
```

Comment:
```suggestion
    hatch env show --json | jq -r ".${ENV_NAME}.dependencies | .[]" | pip-compile -U -o "$TARGET"
```

Assuming that is a mistake.

Reply:
No, this is necessary. It means `pip-compile -U <the result of the pipeline>` while without this you get `pip-compile -U` which defaults to the `default` argument of `pyproject.toml`.

Reply:
Would you prefer `pip-compile - -U -o "$TARGET"`?